pkg/skaffold/deploy/helm.go

```diff
@@ -313,173 +374,139 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
-		if err != nil {
-			return "", errors.Wrapf(err, "cannot parse the image reference %s", tag)
-		}
 
-		var imageTag string
-		if dockerRef.Digest != "" {
-			imageTag = fmt.Sprintf("%s@%s", dockerRef.Tag, dockerRef.Digest)
-		} else {
-			imageTag = dockerRef.Tag
-		}
+	fpath := output[idx+len(tmpDir):]
+	return filepath.Join(tmpDir, fpath), nil
+}
 
-		if cfg.ExplicitRegistry {
-			if dockerRef.Domain == "" {
-				return "", errors.New(fmt.Sprintf("image reference %s has no domain", tag))
-			}
-			return fmt.Sprintf(
-				"%[1]s.registry=%[2]s,%[1]s.repository=%[3]s,%[1]s.tag=%[4]s",
-				valueName,
-				dockerRef.Domain,
-				dockerRef.Path,
-				imageTag,
-			), nil
-		}
-		return fmt.Sprintf(
-			"%[1]s.repository=%[2]s,%[1]s.tag=%[3]s",
-			valueName, dockerRef.BaseName,
-			imageTag,
-		), nil
+// imageSetFromConfig calculates the --set-string value from the helm config
+func imageSetFromConfig(cfg *latest.HelmConventionConfig, valueName string, tag string) (string, error) {
+	if cfg == nil {
+		return fmt.Sprintf("%s=%s", valueName, tag), nil
 	}
-	return fmt.Sprintf("%s=%s", valueName, tag), nil
-}
 
-// Retrieve info about all releases using helm get
-// Skaffold labels will be applied to each deployed k8s object
-// Since helm isn't always consistent with retrieving results, don't return errors here
-func (h *HelmDeployer) getDeployResults(ctx context.Context, namespace string, release string) []Artifact {
-	b, err := h.getReleaseInfo(ctx, release)
+	ref, err := docker.ParseReference(tag)
 	if err != nil {
-		logrus.Warnf(err.Error())
-		return nil
+		return "", errors.Wrapf(err, "cannot parse the image reference %s", tag)
 	}
-	return parseReleaseInfo(namespace, b)
-}
 
-func (h *HelmDeployer) deleteRelease(ctx context.Context, out io.Writer, r latest.HelmRelease) error {
-	releaseName, err := expandTemplate(r.Name)
-	if err != nil {
-		return errors.Wrap(err, "cannot parse the release name template")
+	var imageTag string
+	if ref.Digest != "" {
+		imageTag = fmt.Sprintf("%s@%s", ref.Tag, ref.Digest)
+	} else {
+		imageTag = ref.Tag
 	}
 
-	if err := h.helm(ctx, out, false, "delete", releaseName, "--purge"); err != nil {
-		logrus.Debugf("deleting release %s: %v\n", releaseName, err)
+	if cfg.ExplicitRegistry {
+		if ref.Domain == "" {
+			return "", errors.New(fmt.Sprintf("image reference %s has no domain", tag))
+		}
+		return fmt.Sprintf("%[1]s.registry=%[2]s,%[1]s.repository=%[3]s,%[1]s.tag=%[4]s", valueName, ref.Domain, ref.Path, imageTag), nil
 	}
 
-	return nil
+	return fmt.Sprintf("%[1]s.repository=%[2]s,%[1]s.tag=%[3]s", valueName, ref.BaseName, imageTag), nil
 }
 
-func (h *HelmDeployer) joinTagsToBuildResult(builds []build.Artifact, params map[string]string) (map[string]build.Artifact, error) {
+// assocParamsToArtifact associates parameters to the build artifact it creates
+func assocParamsToArtifact(builds []build.Artifact, params map[string]string) (map[string]build.Artifact, error) {
```

Comment:
nit: I don't love abbreviations in function names, maybe `pairParamsToArtifacts`?